### PR TITLE
adds token set creator and tests

### PR DESCRIPTION
**Pull request by anthonymartin**

### CODE REVIEW

**Review by ecmendenhall — COMMENTED** [review]
One question about interfaces, but this looks great!

I think the next step would be adding a `CustomOracleNavIssuanceModule` and our own `SetValuer` to enable NAV issuance. I think it's possible to add that module to an existing Set, right?

**Review by ecmendenhall — APPROVED** [review]

File: packages/contracts/lib/TokenSet/TokenSetCreator.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  const { targetNAV } = configuration
+    ? configuration
+    : DefaultConfiguration;
+
+  return {
+    _calculateUnits: async function (
+      component: Configuration["components"][0]
+    ): Promise<BigNumber> {
+      const yVault = await hre.ethers.getContractAt(
+        "MockYearnV2Vault",
```
[the remark this answers is not in the document]
Comment (anthonymartin) [reply packages/contracts/lib/TokenSet/TokenSetCreator.ts:36]: yep, you got it. agreed

[the remark this answers is not in the document]
Comment (ecmendenhall) [reply packages/contracts/lib/TokenSet/TokenSetCreator.ts:36]: (Happy to do that in a separate PR—I don't have objections to merging as-is.)

[the remark this answers is not in the document]
Comment (anthonymartin) [reply packages/contracts/lib/TokenSet/TokenSetCreator.ts:36]: ok, I'll write a ticket for it then

**Pull request merged**